source/blog/2022-02-10-announcing-hanami-200alpha6.html.markdown

```diff
@@ -0,0 +1,166 @@
+
+### Providers
+
+First thing first, **we renamed _bootable compontents_ into _providers_**.
+This change was reflected into the public API.
+
+We also changed provider `init` into `prepare`.
+
+```ruby
+Hanami.application.register_provider(:metrics) do
+  prepare do
+    require "datadog/statsd"
+
+    # see application's config/settings.rb
+    @settings = settings.statsd
+  end
+
+  start do
+    register "metrics", Datadog::Statsd.new(@settings.host, @settings.port)
+  end
+
+  stop do
+    # triggered when the application is shut down
+    metrics.close
+  end
+end
+```
+
+```ruby
+module API
+  module Actions
+    module Users
+      class Create < API::Action
+        include Deps["metrics"]
+
+        def handle(req, res)
+          # ...
+          metrics.increment("user.create")
+        end
+      end
+    end
+  end
+end
+```
+
+### Partial Slice Import/Export
+
+A Slice in your application can be used to provide a single responsility functionality.
+Imagine **like a microservices approach in your monolith**, but without the complexity!
+
+For instance, the `search` Slice can expose the search functionalities to other slices.
+
+To ensure unwanted functionalities exports, you have the option of declaring which functionalities to export.
+Here's a few examples
+
+Import from `search` slice, uses `search` as the imported key namespace:
+
+```ruby
+# config/application.rb
+
+module MyApp
+  class Application < Hanami::Application
+    config.slice(:admin) do
+      import(from: :search)
+    end
+  end
+end
+```
+
+Import from `search` slice with custom namespace:
+
+```ruby
+# config/application.rb
+
+module MyApp
+  class Application < Hanami::Application
+    config.slice(:admin) do
+      import(from: :search, as: :search_engine)
+    end
+  end
+end
+```
+
+Import specific keys from `search` slice
+
+```ruby
+# config/application.rb
+
+module MyApp
+  class Application < Hanami::Application
+    config.slice(:admin) do
+      import(keys: ["run_query"], from: :search)
+    end
+  end
+end
+```
+
+Export only specific keys from `search` slice, and import them in `admin`
+
+```ruby
+# config/application.rb
+
+module MyApp
+  class Application < Hanami::Application
+    config.slice(:admin) do
+      import(from: :search)
+    end
+
+    config.slice(:search) do
+      container.config.exports = %w[run_query index_item]
+    end
+  end
+end
+```
+
+## Ruby 3.0+ only
+
+We took this decision for a clear cut with the past of Ruby.
+At the time of the writing (Feb 2022), Ruby 2.6 will reach [End Of Life](https://www.ruby-lang.org/en/downloads/branches/) (EOL) in a month.
+It didn't make sense to still support it.
+
+We want further than that, given the opportunity that we have with Hanami 2 to "start fresh" with the Ruby versions to support.
+We opted for taking 2.7 out as well.
+
+There are a few inconsistencies that have been fixed in Ruby 3.0.
+And we want to get advantage of those.
```

Comment:
This should be a single sentence and probably changed to something like this:

"There are a few inconsistencies that have been fixed in Ruby 3.0 and we want to take advantage of this."